pulsar-testclient/src/main/java/org/apache/pulsar/proxy/socket/client/PerformanceClient.java

```diff
@@ -242,10 +240,19 @@ public void runPerformanceTest(long messages, long limit, int numOfTopic, int si
         executor.submit(() -> {
             try {
                 RateLimiter rateLimiter = RateLimiter.create(limit);
+                long startTime = System.nanoTime();
+                long testEndTime = startTime + (long) (testTime * 1e9);
                 // Send messages on all topics/producers
                 long totalSent = 0;
                 while (true) {
                     for (String topic : producersMap.keySet()) {
+                        if (testTime > 0) {
+                            if (System.nanoTime() > testEndTime) {
+                                log.info("------------- DONE (reached the maximum duration: [{} seconds] of production) --------------", testTime);
```

Comment:
could you please log print the total message numbers. when reach the test time.

Reply:
It has been added in another [pr](https://github.com/apache/pulsar/pull/12016/files#diff-cf545bbba2c55f89cbaf23d8ef206e8ac765b82e2fe0652c3253a23c310f9d36R251), and then merge will be fine.

Reply:
This [pr](https://github.com/apache/pulsar/pull/11985/files#diff-cf545bbba2c55f89cbaf23d8ef206e8ac765b82e2fe0652c3253a23c310f9d36R354) adds the `printAggregatedStats` method and it prints the total number of messages. After this pr is merged, I only need to call the method in line252.

Reply:
We could print throughput and delay data by adding a shutdown hook in other pr([Printing aggregated data when client exit](https://github.com/apache/pulsar/pull/11985/files#diff-cf545bbba2c55f89cbaf23d8ef206e8ac765b82e2fe0652c3253a23c310f9d36R335-R339)), which should meet your requirements.